### DISCUSSION TITLE
What are the sources of this project?

### DISCUSSION THREAD

**Discussion by enrique-lozano** [f1db/f1db#6] [Q&A] [answered]
First of all, thanks for this open project and congratulations on the work.

Just out of curiosity, I would like to know what the sources of the data are, and how they are automatically entered after each race so quickly.

Cheers and thank you again.

**Comment by marceloverdijk** [comment, marked as answer]
Hi Enrique,

I've build up this database over many years. I think I started already with more than 5 years ago as source for some hobby project.

I used many sources for the data. This includes various websites (wikipedias, motorsportmagazine, f1fandom, etc..) and books, but also reaching out to inside people for e.g. missing data online. I used wikipeda a lot for driver/constructor/circuit data, and the references between this data over the years.

Most of the work went into verifying all data as multiple online resources and books had divergent data.
Thinking about it, I should write a blog post or something about how the history of this project.

Timely adding new data is not really an issue/challenge. After every practice, qualifying session, or race this data is published by many websites, or can even be easily retrieved by taking a screenshot of the result on TV ;-)


Cheers,
Marcel

**Comment by enrique-lozano** [comment]
Hello! Thanks for your answer!

I would be delighted to read that post about how you got the data! Thank you very much again, I am working on a formula 1 statistics website, and I may use your database for the project. I'll keep you informed!

Cheers!

**Reply by marceloverdijk** [reply]
Sure , please let me know how it goes. I'm building a website F1 statistics as well ;-)

**Reply by enrique-lozano** [reply]
Of course I will! Just as a suggestion it will be nice to have the data in csv format (takes up less space in disk). Also, for performance reasons and to save space, it would be nice if the IDs were shorter (you could make the id of the countries their ISO code, and for the rest of the IDs implement auto-incrementing numeric IDs, by initials...)

**Reply by marceloverdijk** [reply]
Please create a new feature request [here](https://github.com/f1db/f1db/issues) for the csv.
We can then discuss how the csv should look and I can see what I can do.
